DEVELOPMENT.md

```diff
@@ -26,6 +26,25 @@ To run end-to-end tests you will also need:
 
 Installation instructions for these tools differ heavily based on your operating system and aren't covered here.
 
+### CouchDB on Docker
+
+We recommend using Docker to install and use CouchDB. This ensures you are getting a compatible version and are not relying on OS packages that may be more or less up to date than needed. 
+
+After installing docker, you can create a docker container like so:
+
+```sh
+docker run -d -p 4984:5984 -p 4986:5986 --name medic-couchdb -v /home/scdf/dockers/e2e-couchdb/opt/couchdb/data:/opt/couchdb/data apache/couchdb:2
+```
+
+Notes before copy pasting:
+ - `--name` creates a container called `medic-couchdb`. You can name it whatever you want, but this is how you refer to it later
+ - `-v` maps where couchdb stores data to your local file system for performance, using the path *before* the `:` (the path after the colon is the internal path inside the docker image). This should be somewhere in your home directory you have write access to, and want this data to be stored.
+ - `apache/couchdb:2` will install the latest package for CouchDB 2.x
```

Comment:
After this step I had to go through the clustering setup steps at http://localhost:4984/_utils/#/setup which I think is part of the couchdb installation instructions but worth including here as well.

Reply:
Yeah, I had to do this setup as well, so +1.

and @garethbowen  - were you able to harden the DB (`grunt secure-couchdb`) such that `grunt dev-api` didn't complain about your weak sauce security ("CouchDB security seems to be misconfigured") like it does for me? Just wondering if I'm being helpful in calling this out or if the couchdb newbie is just getting in the way with the PR (feel free to tell me to chill out ;)

Reply:
I had a problem with the node name so the hardening didn't quite work but I haven't had a chance to dig through it yet. I think I set the wrong name when I did the cluster setup.

Reply:
> Just wondering if I'm being helpful in calling this out or if the couchdb newbie is just getting in the way with the PR (feel free to tell me to chill out ;)

This is definitely helpful. It's one of those things that when you've set it up 100 times you can't remember what's not obvious to a first timer! We definitely want to make the setup as painless as possible to help onboarding for new teammates as well as community contributors.

Reply:
> I had a problem with the node name so the hardening didn't quite work but I haven't had a chance to dig through it yet. I think I set the wrong name when I did the cluster setup.

Ok, good to get confirmation on that.  We'll let @SCdF come back with another draft maybe then with explicit steps that cover the hardening w/ docker?  I'll be happy to bring my newbness testing to see if I can break it!

Reply:
w00t! I dug into our [`Gruntfile.js`](https://github.com/medic/cht-core/blob/master/Gruntfile.js) and found the four curl calls that `secure-couchdb` does via [`setup-admin`](https://github.com/medic/cht-core/blob/master/Gruntfile.js#L472).  Combining this with what Fauxton was showing me in the URL for my docker instance, we just need to change how we reference the config db (er, I think that's what it is, still a newb).  Instead of `couchdb@127.0.0.1` we need to use `nonode@nohost` like so:

```
COUCH_URL=http://admin:admin@localhost:4984/medic COUCH_NODE_NAME=nonode@nohost grunt secure-couchdb
```

which works! And then of course we need set our exports accordingly:

```
export COUCH_NODE_NAME=nonode@nohost&&export COUCH_URL=http://admin:admin@localhost:4984/medic
```

and everything false into place with `grunt dev-api` and `grunt dev-sentinel`  and bootstraps up to a functional dev instance.

Reply:
So now we're recommending Docker we can have the node name default to the docker one (instead of I think the default linux one), which will help to streamline this as well.